app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -104,13 +106,30 @@ public void onAttach(Context context) {
     }
 
     @Override
-    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+    public void onCreateOptionsMenu(final Menu menu, MenuInflater inflater) {
         switch (settingsScreen) {
             case SEARCH_ENGINES:
                 inflater.inflate(R.menu.menu_search_engines, menu);
                 break;
             case REMOVE_ENGINES:
                 inflater.inflate(R.menu.menu_remove_search_engines, menu);
+
+                // We disable the trash icon when no engine is checked, enable it otherwise
+                getView().post(new Runnable() {
+                    @Override
+                    public void run() {
+                        final MultiselectSearchEngineListPreference pref = (MultiselectSearchEngineListPreference) getPreferenceScreen()
+                                .findPreference(getResources()
+                                        .getString(R.string.pref_key_multiselect_search_engine_list));
+
+                        if (!pref.isSearchEnginesBindingToMenu()) {
+                            pref.bindEngineCheckboxesToMenu();
+                        }
+
+                        MenuItem menuDeleteItem = menu.findItem(R.id.menu_delete_items);
+                        SettingsFragment.this.setMenuItemEnabled(menuDeleteItem, pref.atLeastOneEngineChecked());
```

Comment:
You can move this "setMenuItemEnabled" line into onPrepareMenuOptions - that way, you don't need to post it as a Runnable on the view, because in onPrepareMenuOptions, the menu items are already created and you can simply call methods on the menuItem itself.

Reply:
Here the problem is not the menuItem but the preference view. If we call the method `atLeastOneEngineChecked` outside the runnable, the activity will crash cause engines checkboxes would be not inflated yet.